### PR TITLE
fixes make error in the plugin system for mac users 

### PR DESCRIPTION
**Pull request by renepickhardt**
I don't know why @brakmic only opened an issue #2090 instead of providing this patch. I believe this enhancement will be useful for any mac user as I just experienced exactly the same problem.

### CODE REVIEW

File: lightningd/plugin.c [hunk 1]
```diff
@@ -9,6 +9,12 @@
 #include <lightningd/json.h>
 #include <unistd.h>
 
+#if defined(__APPLE__) && defined(__MACH__)
```
Comment (ZmnSCPxj) [inline lightningd/plugin.c]: Why not just add all these includes unconditionally?

Comment (renepickhardt) [reply lightningd/plugin.c]: as far as I understood the issue this problem only occurred if macos is used. that is why I added them solely for macos

**Comment by brakmic** [comment]
Hi @renepickhardt 

I simply wasn't sure whether I should provide a patch for macOS that is not officially supported. I saw in their README that macOS *could work* after having applied some extras. And indeed I had to change a few things to make it work on my mac. 

So I thought I'd better avoid opening a Pull Request and let others, more experienced people decide.
But in any case, many thanks for having "escalated" it 😺 

Best,
Harris

**Comment by cdecker** [comment]
For what it's worth the CI failures seem to be only complaining about the order of the header includes, which is expected to be in lexicographic order (except some small exceptions).

**Comment by renepickhardt** [comment]
this means that we have to have several #ifdefs ? or just put the headers in lexecographic order within the #ifdef?

**Comment by conscott** [comment]
Also seems to be a duplicate with #2114  (which includes lexographically, but doesn't ifdef)

**Comment by ZmnSCPxj** [comment]
If it works even on non Mac systems, I suggest to do without `#ifdef` and add in lexicographical order.

**Comment by renepickhardt** [comment]
@arowser sorry for the duplicate PR! I could make a PR on your side include the #if !defined() and then we ask to merge your request.

**Comment by cdecker** [comment]
So with `<signal.h>` included, does this now work on Mac OS?

**Comment by cdecker** [comment]
AFAIK the latest `master` now compiles correctly. Is that true @renepickhardt ?

**Comment by renepickhardt** [comment]
Yes

**Comment by cdecker** [comment]
Thanks for confirming @renepickhardt :+1:

**Pull request closed without merging**